File: Algorithms/1/two-sum.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	nums := []int{5, 75, 25}
	target := 100
	fmt.Println(twoSum(nums, target))
}

func twoSum(nums []int, target int) []int {
	ary := make([]int, len(nums))
	copy(ary, nums)
	sort.Ints(ary)
	lft, rit := 0, len(ary)-1

	for lft < rit {
		tmp := ary[lft] + ary[rit]
		if tmp == target {
			break
		}
		if tmp < target {
			lft++
		} else {
			rit--
		}
	}

	optLft, optRit := 0, 0
	for i, v := range nums {
		if v == ary[lft] || v == ary[rit] {
			optLft = i
			break
		}
	}
	for i, v := range nums[optLft+1:] {
		if v == ary[lft] || v == ary[rit] {
			optRit = optLft + 1 + i
			break
		}
	}

	return []int{optLft, optRit}
}

File: Algorithms/[1, 1k)/1/two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        let ary = nums.sorted()
        var l = 0
        var r = ary.count - 1
        
        while l < r {
            let tmp = ary[l] + ary[r]
            if tmp == target {
                break
            }
            
            if tmp < target {
                l += 1
            } else {
                r -= 1
            }
        }
        
        guard let lft = nums.firstIndex(of: ary[l]) else {
            return []
        }
        guard let rit = nums.lastIndex(of: ary[r]) else {
            return []
        }
        
        return [lft, rit]
    }
}
